api-server/server/component-passport.js

```diff
@@ -81,9 +81,8 @@ export const saveResponseAuthCookies = () => {
 export const loginRedirect = () => {
   return (req, res) => {
     const successRedirect = req => {
-      if (!!req && req.session && req.session.returnTo) {
-        delete req.session.returnTo;
-        return `${homeLocation}/learn`;
+      if (req && req.query && req.query.returnTo) {
+        return req.query.returnTo;
```

Comment:
```suggestion
      if (req && req.session && req.session.returnTo) {
      	delete req.session.returnTo;
      }

      if (req && req.query && req.query.returnTo) {
        return req.query.returnTo;
```

Just a failsafe if the session key ever gets set. We might as well clean it up? What do you think?

Reply:
We do set the session key
https://github.com/freeCodeCamp/freeCodeCamp/blob/master/api-server/server/middlewares/add-return-to.js

but we don't use it.

There's no harm in deleting it, so sure, but we should remove or use that middleware, too.

Reply:
Right that whole thing should go.

Reply:
I checked there was nothing worth salvaging from it (there wasn't) and took it out.